libraries/AP_Compass/AP_Compass_MMC3416.cpp

```diff
@@ -102,6 +108,9 @@ bool AP_Compass_MMC3416::init()
     dev->register_periodic_callback(10000,
                                     FUNCTOR_BIND_MEMBER(&AP_Compass_MMC3416::timer, bool));
 
+    // wait 250ms for the compass to make it's initial readings
+    hal.scheduler->delay(250);
```

Comment:
if you were waiting and doing something.... ok, but just wait, why?  Isn't this just going to slowdown the startup sequence which is serialized?